SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/GridBasedLBStrategy.cpp

```diff
@@ -100,6 +101,29 @@ VirtualWorkerId UGridBasedLBStrategy::WhoShouldHaveAuthority(const AActor& Actor
 	return SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
 }
 
+SpatialGDK::QueryConstraint UGridBasedLBStrategy::GetWorkerInterestQueryConstraint() const
+{
+	// For a grid-based strategy, the interest area is the cell that the worker is authoritative over plus some border region.
+	check(IsReady());
+	check(InterestBorder >= 0);
+	check(Rows * Cols > 0);
```

Comment:
Maybe this check is overly defensive, if rows * cols = 0 then the strategy will return no valid virtual worker IDs meaning the IsReady() should never be set
```suggestion
```

Reply:
How is the local virtual worker ID flow related to the rows and cols? It seems the strategy could have a local virtual worker ID set and announce that it is ready regardless of how many virtual worker IDs there should be. Is it because a worker will never register itself with a virtual worker ID if the strategy is invalid?

Reply:
virtual worker translation manager gets a list of unassigned virtual workers from the strategy. If rows * cols = 0 then the strategy will return 0 virtual worker IDs meaning the local virtual worker ID will never be set, meaning IsReady() never returns true

Reply:
cool removed